c7n/resources/sns.py

```diff
@@ -325,3 +325,36 @@ def process(self, resources):
                 AttributeValue=key
             )
         return resources
+
+
+@SNS.action_registry.register('delete')
+class DeleteTopic(BaseAction):
+    """
+    Deletes a SNS Topic
+
+    :example:
+
+    .. code-block:: yaml
+
+      policies:
+        - name: delete-bad-topic
+          resource: aws.sns
+          filters:
+            - TopicArn: arn:aws:sns:us-east-2:123456789012:BadTopic
+          actions:
+            - type: delete
+    """
+
+    schema = type_schema('delete')
+
+    permissions = ('sns:DeleteTopic',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('sns')
+        for r in resources:
+            try:
+                client.delete_topic(TopicArn=r['TopicArn'])
+            except client.exceptions.NotFoundException:
+                continue
+            else:
```

Comment:
this else clause can be dropped, if not caught by the except it will raise. try/except/else goes to the else clause if there's not something thats raised in the try/except.. ie this doesn't make sense, and i'm a bit surprised it passed.

https://docs.python.org/3/tutorial/errors.html#handling-exceptions

```python
>>> def foo(exc):
...    try:
...      if exc:
...        raise exc
...    except ValueError:
...      print('caught')
...    else:
...      print('woot')
...      raise
...
>>> foo(ValueError())
caught
>>> foo(SyntaxError())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 4, in foo
SyntaxError: None
>>> foo(None)
woot
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 9, in foo
RuntimeError: No active exception to reraise
```